// lp-dialogs/esm2015/lib/lp-info-dialog/lp-info-dialog.component.js
import { Component, Inject } from '@angular/core';
import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
export class LpInfoDialogComponent {
    constructor(dialogRef, data) {
        this.dialogRef = dialogRef;
        this.data = data;
    }
    ngOnInit() {
        this.message = this.data.message;
        this.title = this.data.title;
        this.item = this.data.item;
    }
    hideModal() {
        this.dialogRef.close();
    }
    /// MUST CREATE A BACKEND TO REPORT ERRORS ///
    reportError() { }
}
LpInfoDialogComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-lp-info-dialog',
                template: "<h2 mat-dialog-title> {{title}} </h2>\n<mat-dialog-content>\n    <p >\n        {{item}}\n    </p>\n    <div class=\"div-message\" >\n       <p>{{message}}</p>\n    </div>\n</mat-dialog-content>\n<mat-dialog-actions>\n    <button class=\"mat-raised-button\" (click)=\"hideModal()\">Close</button>\n</mat-dialog-actions>\n",
                styles: [".div-message{max-width:35vw}.div-message p{text-align:center;word-break:break-all}"]
            },] }
];
LpInfoDialogComponent.ctorParameters = () => [
    { type: MatDialogRef },
    { type: undefined, decorators: [{ type: Inject, args: [MAT_DIALOG_DATA,] }] }
];
