// lp-dialogs/esm2015/lib/shared/providers/lp-dialogs.service.js
import { Injectable } from '@angular/core';
import { MatDialog, MatDialogConfig } from '@angular/material/dialog';
import { LpInfoDialogComponent } from '../../lp-info-dialog/lp-info-dialog.component';
import { LpConfirmDialogComponent } from '../../lp-confirm-dialog/lp-confirm-dialog.component';
import { LpDialogsModule } from '../../lp-dialogs.module';
import * as i0 from "@angular/core";
import * as i1 from "@angular/material/dialog";
import * as i2 from "../../lp-dialogs.module";
export class LpDialogsService {
    constructor(dialog) {
        this.dialog = dialog;
    }
    openConfirmDialog(title = 'CONFIRM DELETION', message = 'Are you sure?') {
        const dialog = new MatDialogConfig();
        dialog.disableClose = true;
        dialog.autoFocus = true;
        dialog.data = { title, message };
        let dialogRef = this.dialog.open(LpConfirmDialogComponent, dialog);
        return dialogRef.afterClosed();
    }
    openInfoDialog(message, title, item) {
        const dialog = new MatDialogConfig();
        dialog.disableClose = true;
        dialog.autoFocus = true;
        dialog.data = { message, title, item: item ? item : null };
        let dialogRef = this.dialog.open(LpInfoDialogComponent, dialog);
        return dialogRef.afterClosed();
    }
}
LpDialogsService.ɵprov = i0.ɵɵdefineInjectable({ factory: function LpDialogsService_Factory() { return new LpDialogsService(i0.ɵɵinject(i1.MatDialog)); }, token: LpDialogsService, providedIn: i2.LpDialogsModule });
LpDialogsService.decorators = [
    { type: Injectable, args: [{
                providedIn: LpDialogsModule
            },] }
];
LpDialogsService.ctorParameters = () => [
    { type: MatDialog }
];
